tinygrad/codegen/linearizer.py

```diff
@@ -381,10 +389,13 @@ def linearize(self):
     acc: List[UOp] = []
     self.load_cache: Dict[str, UOp] = {}
 
-    # reduce op
+    # reduce ops
+    assert len(self.reduceops) == len(set(self.reduceops)), "All reduceops must be unique"
+    self.reduceops.reverse()
```

Comment:
yeah you need to linearize the deepest reduceop first.

I think .reverse() won't work for multi output because you could have an AST like:
```
  0 ━┳ STORE MemBuffer(idx=0, dtype=dtypes.float, st=ShapeTracker(views=(View(shape=(1,), strides=(0,), offset=0, mask=None, contiguous=True),)))
  1  ┗━┳ SUM ((0,), dtypes.float)
  2    ┗━━ LOAD MemBuffer(idx=2, dtype=dtypes.float, st=ShapeTracker(views=(View(shape=(32,), strides=(1,), offset=0, mask=None, contiguous=True),)))
  0 ━┳ STORE MemBuffer(idx=1, dtype=dtypes.float, st=ShapeTracker(views=(View(shape=(1,), strides=(0,), offset=0, mask=None, contiguous=True),)))
  1  ┗━┳ SUM ((0,), dtypes.float)
  2    ┗━┳ SUB 
  3      ┣━━ LOAD MemBuffer(idx=2, dtype=dtypes.float, st=ShapeTracker(views=(View(shape=(32,), strides=(1,), offset=0, mask=None, contiguous=True),)))
  4      ┗━┳ SUM ((0,), dtypes.float)
  5        ┗━━ LOAD MemBuffer(idx=2, dtype=dtypes.float, st=ShapeTracker(views=(View(shape=(32,), strides=(1,), offset=0, mask=None, contiguous=True),)))
```

two ideas:
- self.reduceops orders by depth
- `render_reduceop` recurses (similar to parse_ast)